File: Python/Next Greater Element I.py
class Solution(object):
    def nextGreaterElement(self, findNums, nums):
        """
        :type findNums: List[int]
        :type nums: List[int]
        :rtype: List[int]
        """
        self.nums = nums
        return map(self.next_num_in_nums, findNums)
        
    def next_num_in_nums(self, num):
        try:
            index = self.nums.index(num) + 1
            for n in self.nums[index:]:
                if n > num:
                    return n
            else:
                return -1
        except:
            return -1

File: cpp/Next Greater Element I.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& findNums, vector<int>& nums) {
        vector<int> result;

        for (int find_num : findNums) {
            vector<int>::iterator index = find(nums.begin(), nums.end(), find_num);
            bool find_flag = false;
            for (vector<int>::iterator i = index; i < nums.end(); ++i) {
                if (*i > find_num) {
                    result.push_back(*i);
                    find_flag = true;
                    break;
                }
            }
            if (!find_flag) {
                result.push_back(-1);
            }
        }
        return result;
    }
};
